chronos/transports/common.nim

```diff
@@ -625,3 +631,72 @@ when defined(windows):
        {.stdcall, dynlib: "kernel32", importc: "SetNamedPipeHandleState".}
   proc resetEvent*(hEvent: HANDLE): WINBOOL
        {.stdcall, dynlib: "kernel32", importc: "ResetEvent".}
+
+template getTransportTooManyError*(code: int = 0): ref TransportTooManyError =
+  let msg =
+    when defined(posix):
+      if code == 0:
+        "Too many open transports"
+      elif code == EMFILE:
+        "[EMFILE] Too many open files in the process"
+      elif code == ENFILE:
+        "[ENFILE] Too many open files in system"
+      elif code == ENOBUFS:
+        "[ENOBUFS] No buffer space available"
+      elif code == ENOMEM:
+        "[ENOMEM] Not enough memory availble"
+      else:
+        "[" & $code & "] Too many open transports"
+    elif defined(windows):
+      case code
+      of 0:
+        "Too many open transports"
+      of ERROR_TOO_MANY_OPEN_FILES:
+        "[ERROR_TOO_MANY_OPEN_FILES] Too many open files"
+      of WSAENOBUFS:
+        "[WSAENOBUFS] No buffer space available"
+      of WSAEMFILE:
+        "[WSAEMFILE] Too many open sockets"
+      else:
+        "[" & $code & "] Too many open transports"
+    else:
+      "[" & $code & "] Too many open transports"
+  newException(TransportTooManyError, msg)
+
+template getConnectionAbortedError*(m: string = ""): ref TransportAbortedError =
+  let msg =
+    if len(m) == 0:
+      "[ECONNABORTED] Connection has been aborted before being accepted"
+    else:
+      "[ECONNABORTED] " & m
+  newException(TransportAbortedError, msg)
+
+template getConnectionAbortedError*(code: int): ref TransportAbortedError =
+  let msg =
+    when defined(posix):
+      if code == 0:
+        "[ECONNABORTED] Connection has been aborted before being accepted"
+      elif code == EPERM:
+        "[EPERM] Firewall rules forbid connection"
```

Comment:
I have seen `EPERM` without firewalls on some obscure TCP stack before, so it _could_ also be other reasons, but I like the general idea of trying to give some context about what could have gone wrong.

The POSIX spec is more generic here: "A protocol error has occurred; [OB XSR] [Option Start]  for example, the STREAMS protocol stack has not been initialized. [Option End]"
https://pubs.opengroup.org/onlinepubs/9699919799/functions/accept.html